### ISSUE TITLE
A way to decrease time to get back an "Error" shown on HUD

### ISSUE THREAD

**Issue by foffux** [kevin-hirsch/KVNProgress#40]
Hello @kevin-hirsch .
I want to ask for a question. When I use tour HUD in my apps, it could happen that network connection is not good or even not present at all. In these scenarios, since I configured an infinite loop circle, app could stop and wait for a connection error response for a long time. My question is: is there any setup I could use to decrease the maximum connection time I will wait for before telling user communication is down?
In other words: is there any change to configure your HUD to get back an "Error" shown on it?
I hope my question is clear enough for you.
Let me know.
Thank you.

**Comment by kevin-hirsch** [comment]
Hi @foffux,

No, there is no network connection helper for the moment.
But what you could do is:
- check if internet connection is available before executing your request and showing the HUD (http://stackoverflow.com/a/3597085/2571566)
- specify a maximum time to wait for server response to avoid too long calls and show an error if this time is reached without any response from the server. (also called timeout or `timeoutIntervalForRequest ` in iOS on `NSURLSessionConfiguration`: [documentation here](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSessionConfiguration_class/index.html#//apple_ref/occ/instp/NSURLSessionConfiguration/timeoutIntervalForRequest))

**Comment by foffux** [comment]
Hello.
I did what you said. I think, at least at the moment, it is okay for me.
Thank you for your support.

2015-05-13 10:30 GMT+02:00 Kevin Hirsch <notifications@github.com>:

**Comment by kevin-hirsch** [comment]
Hello! I'm glad to help :)
You're welcome.

I'm closing this issue.

**Comment by gavrilaady** [comment]
Hi...can you help me?..i run pod 'KVNProgress',then i imported KVNprogress to my ViewController but i got an error:'No such module 'KVNProgress'' although i can use KVNProgress.show()(for example) in my code....any suggestions?
